sfge/src/main/java/com/salesforce/graph/build/AbstractApexVertexBuilder.java

```diff
@@ -84,20 +105,44 @@ private void buildVertices(JorjeNode node, Vertex vNodeParam, String fileName) {
             if (StaticBlockUtil.isStaticBlockStatement(node, child)) {
                 final Vertex parentVertexForChild =
                         StaticBlockUtil.createSyntheticStaticBlockMethod(g, vNode, i);
+                // The child vertex is made a child of the synthetic vertex instead of its original
+                // parent.
                 GremlinVertexUtil.addParentChildRelationship(g, parentVertexForChild, vChild);
                 verticesAddressed.add(parentVertexForChild);
             } else {
                 GremlinVertexUtil.addParentChildRelationship(g, vNode, vChild);
             }
 
-            if (vPreviousSibling != null) {
-                g.addE(Schema.NEXT_SIBLING).from(vPreviousSibling).to(vChild).iterate();
-            }
-            vPreviousSibling = vChild;
+            // Map this vertex as the Nth child vertex of the parent.
+            // TODO: This might be a bug. It's possible we should actually be using the synthetic
+            //       vertex if we create one. Do more investigating around that!
+            childVerticesByIndex.put(child.getChildIndex(), vChild);
 
             // To save memory in the graph, don't pass the source name into recursive calls.
             buildVertices(child, vChild, null);
         }
+
+        // If any child vertices were created, they need to be properly linked as siblings.
+        if (!childVerticesByIndex.isEmpty()) {
+            Vertex vLastChild = null;
+            int linkedChildren = 0;
+            int childCount = childVerticesByIndex.size();
+            int i = 0;
+            while (linkedChildren < childCount) {
```

Comment:
nit: personally, i prefer foreach or for loop variant as it lends itself for easier reading.  leave that to you.  I wouldn't be sad with while loop :)

Reply:
I agree, but a `while` loop was the best option here, since the number of children might not necessarily cleanly correspond to the indices.